lib/mongodb.js

```diff
@@ -1034,7 +1059,7 @@ MongoDB.prototype.all = function all(model, filter, options, callback) {
   var fields = filter.fields;
 
   // Convert custom column names
-  fields = self.fromPropertyToDatabaseNames(model, fields);
+  fields = self.fromFieldFilterToDatabaseNames(model, fields);
```

Comment:
Why not replace the functionality of `fromPropertyToDatabaseNames` instead of making a new prototype method?

Reply:
The function "fromPropertyToDatabaseNames" is used elsewhere in the connector to convert the data from the loopback model to the database schema on saving data and vice versa on reading data. 

The Loopback fields filter gives me an array with fields to filter, this array is not supported by the "fromPropertyToDatabseNames" (treats it like an object which breaks the filter). Because of that I wrote a new function that handles the array correctly.